src/Chapter1.hs

```diff
@@ -554,7 +556,14 @@ value after "=" where the condition is true.
 Casual reminder about adding top-level type signatures for all functions :)
 -}
 
-mid x y z = error "mid: not implemented!"
+mid :: Int -> Int -> Int -> Int
+mid x y z
+    | x >= y && y >= z = y
+    | z >= y && y >= z = y
+    | y >= x && x >= z = x
+    | z >= x && x >= y = x
+    | x >= z && z >= y = z
+    | y >= z && z >= x = z
```

Comment:
As we mentioned, the compiler in Haskell is very attentive to the exhaustive pattern-matching. And here it would warn you that `Pattern matching is not exhaustive`, as the guards have quite complicated logic, and the compiler won't be able to prove that it covers all the cases.

Because of that, you will need to add another guard – `| otherwise = ...`, to tell the compiler, that your pattern matching is exhaustive 🙂

But as you correctly noticed, you have covered all possible combinations.
That meant that when pattern matching would come to these two cases, the answer will always be `z`. So in reality, you don't need to check these two conditions:

```suggestion
    | otherwise = z
```